soft_delete.go

```diff
@@ -82,6 +82,24 @@ func (sd SoftDeleteQueryClause) ModifyStatement(stmt *Statement) {
 		}})
 		stmt.Clauses["soft_delete_enabled"] = clause.Clause{}
 	}
+
+	// Modify for Joins[i].ON exprs
+	if _, ok := stmt.Clauses["soft_delete_join_enabled"]; !ok && !stmt.Statement.Unscoped {
```

Comment:
It doesn't seem like a good way to update an join exprs here,  but it fits the function name and doesn't break the changes.
If break changes are allowed, i think we can create a new interface for custom `QueryClauses` to get right `Joins.ON.Exprs`